dlp/snippets/deid_cloud_storage.py

```diff
@@ -0,0 +1,234 @@
+    # See https://cloud.google.com/dlp/docs/infotypes-reference for complete list of info types.
+    inspect_config = {
+        "info_types": [{"name": info_type} for info_type in info_types]
+    }
+
+    # Construct cloud_storage_options dictionary with the bucket's URL.
+    storage_config = {
+        "cloud_storage_options": {
+            "file_set": {
+                "url": f"gs://{input_gcs_bucket}"
+            }
+        }
+    }
+
+    # Specify the big query table to store the transformation details.
+    big_query_table = {
+        "project_id": project,
+        "dataset_id": dataset_id,
+        "table_id": table_id,
+    }
+
+    # Convert the project id into a full resource id.
+    parent = f"projects/{project}"
+
+    # Construct Transformation Configuration with de-identify Templates used
+    # for transformation.
+    transformation_config = {
+        "deidentify_template": f"{parent}/deidentifyTemplates/{deid_template_id}",
+        "structured_deidentify_template": f"{parent}/deidentifyTemplates/{structured_deid_template_id}",
+        "image_redact_template": f"{parent}/deidentifyTemplates/{image_redact_template_id}",
+    }
+
+    # Tell the API where to send notification when the job is completed.
+    actions = [
+        {
+            "deidentify": {
+                "cloud_storage_output": f"gs://{output_gcs_bucket}",
+                "transformation_config": transformation_config,
+                "transformation_details_storage_config": {
+                    "table": big_query_table
+                },
+                "file_types_to_transform": ["IMAGE", "CSV", "TEXT_FILE"],
+            }
+        }
+    ]
+
+    # Construct the job definition.
+    inspect_job = {
+        "inspect_config": inspect_config,
+        "storage_config": storage_config,
+        "actions": actions,
+    }
+
+    # Call the API.
+    response = dlp.create_dlp_job(
+        request={
+            "parent": parent,
+            "inspect_job": inspect_job,
+        }
+    )
+
+    job_name = response.name
+    print(f"Inspection Job started : {job_name}")
+
+    # Waiting for the job to get completed.
+    job = dlp.get_dlp_job(request={"name": job_name})
+    # Since the sleep time is kept as 30s, number of calls would be timeout/30.
+    no_of_attempts = timeout//30
+    while True:
+        # Check if the job has completed.
+        if job.state == google.cloud.dlp_v2.DlpJob.JobState.DONE:
+            break
+        if job.state == google.cloud.dlp_v2.DlpJob.JobState.FAILED:
+            print('Job Failed, Please check the configuration.')
+            return
+
+        # Sleep for a short duration before checking the job status again.
+        time.sleep(30)
+        no_of_attempts -= 1
+
+        # Get DLP job status.
+        job = dlp.get_dlp_job(request={"name": job_name})
+
+    if job.state != google.cloud.dlp_v2.DlpJob.JobState.DONE:
+        print(f"Job did not complete within {timeout} minutes.")
```

Comment:
I think you need to fix the loop above a bit. Currently, there are only 2 stop conditions: the job is DONE or it FAILED. If it failed, we return, so we won't land here.

Reply:
Thank you for your comment, this is my miss, I have updated things now.